packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -363,12 +352,42 @@ where
             &dst_client_id,
             &src_packet_seqs,
             &dst_packet_seqs,
-            &target_height,
             &self.signer_address,
             now_since_unix.as_secs(),
         );
 
-        let mut ethereum_client_state = self.ethereum_client_state(dst_client_id.clone()).await?;
+        let max_timeout_slot = timeout_msgs
```

Comment:
Since all you really need in the end is the minimum_block_number, would it not be possible to create a single chain for all this?

Reply:
The async request in the chain makes this complex. It might be possible but not as simple as merging them to a single chain. Can revisit this in future refactors